### DISCUSSION TITLE
*Editor* Inconsistent Menu Layout

### DISCUSSION THREAD

**Discussion by gomi-bear** [ppy/osu#23610] [Ideas]
### I made a post almost a month ago about this on the Developer forum. I was told I am supposed to upload suggestions and issues here instead, So I'm just going to repost my 2 comments here. If you want to check out the original topic, here's the link https://osu.ppy.sh/community/forums/topics/1754644?n=1 

---------------------------------------------------------------------------------------------------------------------------------------------------

**(Comment 1)**
I’m not sure if “Inconsistent Menu Layout” is the best way of phrasing it, but every time I accidentally open the Editor and I try to quit, I always end up making the same mistake.

I click “Oops, continue editing” Instead of “Forget all changes”.

![68747470733a2f2f696d616765732d7769786d702d6564333061383662386334636138383737373335393463322e7769786d702e636f6d2f662f32376235356431382d373437382d343133612d613266662d3033303034353532376364642f646676343878392d35303064](https://github.com/ppy/osu/assets/39307489/50cc4cd5-8770-4618-8ad9-a7b29b42c0ec)

When I try to quit something in a game or a program and I am prompted with a list, I am going to assume “Quit” at the bottom, Hence why I keep clicking “Oops, continue editing”.

In most games quit is at the bottom and resume is on top, or at least above quit, and this includes Osu! Lazer.

![68747470733a2f2f696d616765732d7769786d702d6564333061383662386334636138383737373335393463322e7769786d702e636f6d2f662f32376235356431382d373437382d343133612d613266662d3033303034353532376364642f64667634386f332d38363636](https://github.com/ppy/osu/assets/39307489/d5b97606-086b-4a3d-b2e7-a1e4717a023e)

If you play a map and pause, you'll see “Quit” is at the bottom and “Continue” is at the top. But if you go to the Editor, the order is reversed. “Continue” is bottom and “Quit” is both in the middle and at the top.

Which doesn’t seem very consistent or intuitive.

If you were to click “esc” on your keyboard, or navigate your way to [Setting → Exit], you’re probably trying to quit the Editor. And I think it would be cleaner and clearer if you were just prompted with two options: “Forget all changes”, and “Save my masterpiece!”.

If you didn’t intend to quit the Editor you could either click “esc”, or just click off the Selection screen. I don’t think that “Oops, continue editing” needs to be there, but if you really want to keep it, you should at least make it clearly separated from the other two options. Or at least put it on top of the list.

---------------------------------------------------------------------------------------------------------------------------------------------------

**(Comment 2)**
I don’t understand why the button should be there, since you can press “esc” on your keyboard, or just click off screen, the sides basically function as two massive “Opps, continue editing” buttons.

This is also how it works in the Beatmap listing. If you’re downloading beatmaps, the only way to close a map you’re looking at, and return to the beatmap listing, is to press “esc” or click off screen.

![68747470733a2f2f696d616765732d7769786d702d6564333061383662386334636138383737373335393463322e7769786d702e636f6d2f662f32376235356431382d373437382d343133612d613266662d3033303034353532376364642f6466766976726b2d34326164](https://github.com/ppy/osu/assets/39307489/ff0a1277-f222-4ee4-86f7-777f38afb821)

This would fix my complaint about the “Opps, continue editing”, and it would make the menu look cleaner.

But I'm fine with the “Opps, continue editing” button being there, as long as it’s clearly separated from the other 2 buttons.

Since “Save my masterpiece!” and “Forget all changes” basically means the same thing, as in they quit the Editor. Contra “Opps, continue editing”, which means the complete opposite.

Usually, not having some obvious separation between the two, isn’t such a big problem since most, or at least a massive amount of games and programs, use the same layout when it comes to where the “Quit” and “Resume” buttons should go. “Quit” goes bottom, and “Resume” goes on top, or at least above “Quit”.

This is also the case in Osu! Lazer, and not just in the Play menu.

In the Editor itself, if you try to quit through [Setting→Exit] in the drop down menu “Exit” is at the bottom.

![68747470733a2f2f696d616765732d7769786d702d6564333061383662386334636138383737373335393463322e7769786d702e636f6d2f662f32376235356431382d373437382d343133612d613266662d3033303034353532376364642f64667669766f6b2d64613736](https://github.com/ppy/osu/assets/39307489/2ec439db-a3d3-42ad-abc4-9690205b3cb6)

If you try to Exit your account “Sign out” is at the bottom of the list.

![68747470733a2f2f696d616765732d7769786d702d6564333061383662386334636138383737373335393463322e7769786d702e636f6d2f662f32376235356431382d373437382d343133612d613266662d3033303034353532376364642f646676697676362d62306636](https://github.com/ppy/osu/assets/39307489/dd3216cc-047b-40a5-8d45-7788189c6326)

Having exit be at the bottom and resume at the top, is the common norm, you’ll see it everywhere.

I’m sure you can get used to resume being at the bottom for this one menu. But for new users it’s not going to be very intuitive, since most applications have it the other way around, including Osu! Lazer.

**Comment by bdach** [comment]
I think the item should still continue to be there, so that new users don't need to figure out that clicking outside of the dialog will cancel the dialog.

No personal opinion on the order of the buttons though. Maybe we could make the dangerous option last.

**Comment by monochrome22** [comment]
> I think the item should still continue to be there, so that new users don't need to figure out that clicking outside of the dialog will cancel the dialog.

Agreed.

I agree with changing the order. In my experience, whenever I try to quit the editor, I have to take a second to read all the texts to make sure I'm pressing the right thing. Now that I think about it, one reason for this might be the ordering. The other reason might be that both save and exit have similar colors. Another reason might be that the phrase "Forget all changes" doesn't include the words "Quit" or "Exit", so I can't click it intuitively, but rather I have to think to make sure it does what I want it to do.

The order should be:
`Oops, continue editing`
`Save my masterpiece!`
`Forget all changes`
It might seem like putting the save button in the middle makes it less important, but that's fine, because if someone hits Esc, then most likely they don't want to save, but rather they either hit Esc by accident or they hit it hoping it will do something else, like move them to the previous screen, or they just want to leave.
